lib/compiler-singlepass/src/codegen.rs

```diff
@@ -6611,6 +6611,8 @@ impl<'a, M: Machine> FuncGen<'a, M> {
         let mut body = self.machine.assembler_finalize()?;
         body.shrink_to_fit();
 
+        save_assembly_to_file("-module-dump.o", &body);
```

Comment:
Remove me before merging, use similar strategy as LLVM debug dir

Reply:
Sure, I removed it and added a tracking issue for the feature: #5833.